### ISSUE TITLE
r/aws_s3_bucket_ownership_controls: support BucketOwnerEnforced

### ISSUE THREAD

**Issue by roberth-k** [hashicorp/terraform-provider-aws#21980]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add support for the `BucketOwnerEnforced` choice of object ownership, which is used to disable ACL-s.

It seems like this should be automatically covered by [`s3.ObjectOwnership_Values()`](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/s3/bucket_ownership_controls.go#L43) once the AWS Go SDK is updated to v1.42.16 or above, so this is just a documentation update.

### New or Affected Resource(s)

* aws_s3_bucket_ownership_controls

### Potential Terraform Configuration

```hcl
resource "aws_s3_bucket_ownership_controls" "example" {
  bucket = aws_s3_bucket.example.id

  rule {
    object_ownership = "BucketOwnerEnforced"
  }
}
```

### References

* https://aws.amazon.com/about-aws/whats-new/2021/11/amazon-s3-object-ownership-simplify-access-management-data-s3/
* https://docs.aws.amazon.com/AmazonS3/latest/API/API_OwnershipControlsRule.html

**Comment by mattburgess** [comment]
Requires aws-sdk-go v1.42.16 (https://github.com/aws/aws-sdk-go/blob/main/CHANGELOG.md#release-v14216-2021-11-30)

**Comment by mattburgess** [comment]
@roberth-k I really should have read _all_ of your description first! The [API documentation](https://pkg.go.dev/github.com/aws/aws-sdk-go/service/s3#CreateBucketInput) shows that one can specify the object ownership directly on the `aws_s3_bucket` resource itself, so that's what I ended up implementing. I'm wondering whether that's the right thing to do or not; technically it matches the API but it does mean there's now 2 ways of doing things and I think the general consensus is to favour the separated out resources vs. bundling them all together. I guess I'll see once a reviewer comes by to take a look!

**Comment by breml** [comment]
Bump to AWS SDK v1.42.17 has already been merged in #22001. I assume, that `BucketOwnerEnforced` will already work with the `aws_s3_bucket_ownership_controls` resource. So the only thing missing is the update of the documentation of the `aws_s3_bucket_ownership_controls` resource.

**Comment by breml** [comment]
I confirm that 

```terraform
resource "aws_s3_bucket_ownership_controls" "example" {
  bucket = aws_s3_bucket.example.id

  rule {
    object_ownership = "BucketOwnerEnforced"
  }
}
```

works with release v3.68.0.
